ytcc/database.py

```diff
@@ -157,25 +158,30 @@ def increase_extractor_fail_count(self, e_hash, max_fail=(1 << 63) - 1) -> None:
                 SET failure_count = failure_count + 1
                 WHERE failure_count < :max_fail
         """
-        self.connection.execute(query, {"e_hash": e_hash, "max_fail": max_fail})
+        with self.connection:
+            self.connection.execute(query, {"e_hash": e_hash, "max_fail": max_fail})
 
     def close(self) -> None:
         self.connection.commit()
         self.connection.close()
 
     def add_playlist(self, name: str, url: str) -> None:
         query = "INSERT INTO playlist (name, url) VALUES (?, ?);"
-        self.connection.execute(query, (name, url))
+        with self.connection:
+            res = self.connection.execute(query, (name, url))
+            return res.rowcount > 0
 
     def delete_playlist(self, name: str) -> bool:
         query = "DELETE FROM playlist WHERE name = ?"
-        res = self.connection.execute(query, (name,))
+        with self.connection:
+            res = self.connection.execute(query, (name,))
         return res.rowcount > 0
```

Comment:
Is it a problem that `res.rowcount` is accessed outside the context manager? Isn't the `res` cursor object closed at this point?

Reply:
Oh yeah, that's a good point. The only reason the `with` block will be exited without the return is if an exception is raised, and we don't care about the return value in that case anyway.